### PR TITLE
Data for (new) stage 3 array-grouping

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Summary

https://github.com/mdn/content/issues/28279

### CODE REVIEW

File: javascript/builtins/Map.json [hunk 1]
```diff
@@ -473,6 +473,74 @@
...
+                  "flags": [
+                    {
+                      "type": "preference",
+                      "name": "javascript.options.experimental.array_grouping",
+                      "value_to_set": "true"
+                    }
+                  ],
+                  "alternative_name": "Array.prototype.groupToMap"
+                }
+              ],
```
Comment (hamishwillee) [inline javascript/builtins/Map.json:512]: The gods of BCD might take a different view, but I see no value in keeping info about the removed preferenced API in FF

```suggestion
              "firefox": {
                "version_added": "preview",
                "flags": [
                  {
                    "type": "preference",
                    "name": "javascript.options.experimental.array_grouping",
                    "value_to_set": "true"
                  }
                ]
              },
```

File: javascript/builtins/Map.json [hunk 2]
```diff
@@ -473,6 +473,74 @@
...
+              "oculus": "mirror",
+              "opera": "mirror",
+              "opera_android": "mirror",
+              "safari": [
+                {
+                  "version_added": "preview"
+                },
+                {
+                  "version_added": "16.4",
+                  "version_removed": "preview",
```
Comment (hamishwillee) [inline javascript/builtins/Map.json:529]: I "think" thisneeds to be a specific version.

File: javascript/builtins/Object.json [hunk 3]
```diff
@@ -823,6 +823,74 @@
...
+                  "flags": [
+                    {
+                      "type": "preference",
+                      "name": "javascript.options.experimental.array_grouping",
+                      "value_to_set": "true"
+                    }
+                  ],
+                  "alternative_name": "Array.prototype.group"
+                }
+              ],
```
Comment (hamishwillee) [inline javascript/builtins/Object.json:862]:
```suggestion
              "firefox": {
                "version_added": "preview",
                "flags": [
                  {
                    "type": "preference",
                    "name": "javascript.options.experimental.array_grouping",
                    "value_to_set": "true"
                  }
                ]
              },
```

File: javascript/builtins/Object.json [hunk 4]
```diff
@@ -823,6 +823,74 @@
...
+              "oculus": "mirror",
+              "opera": "mirror",
+              "opera_android": "mirror",
+              "safari": [
+                {
+                  "version_added": "preview"
+                },
+                {
+                  "version_added": "16.4",
+                  "version_removed": "preview",
```
Comment (hamishwillee) [inline javascript/builtins/Object.json:879]: Actual version

File: javascript/builtins/Map.json (hunk 1 above)
Comment (hamishwillee) [reply javascript/builtins/Map.json:512]: FYI @queengooborg @Elchi3 or whoever reviews this, the preference info here is correct - it's a preference that only exists in whatever the nightly build is, and which is false by default.

**Review by danilodj4242 — APPROVED** [review]

File: javascript/builtins/Map.json (hunk 2 above)
Comment (Josh-Cena) [reply javascript/builtins/Map.json:515]: It's not released yet. The browsers I have access to are 16.5 and TP and it's in TP, but I don't know if it's in 17.0.

Comment (hamishwillee) [reply javascript/builtins/Map.json:515]: Ah, thanks. I don't know what BCD does for this case. Maybe exactly this.

**Review by Elchi3 — APPROVED** [review]
Thanks! 👍

File: javascript/builtins/Map.json [hunk 5]
```diff
@@ -473,6 +473,60 @@
...
+                  "alternative_name": "Array.prototype.groupToMap"
```
Comment (queengooborg) [inline javascript/builtins/Map.json:516]: After-merge review: `alternative_name` isn't really intended to be used this way.  These should be replaced with a note and `partial_implementation`.

**Pull request merged**